src/python/pants/backend/python/dependency_inference/parse_python_imports.py

```diff
@@ -17,11 +17,15 @@
 from pants.util.logging import LogLevel
 
 
-class ParsedPythonImports(FrozenDict[str, int]):
-    """All the discovered imports from a Python source file mapped to the first line they appear.
+@dataclass(frozen=True)
+class ParsedPythonImportInfo:
+    lineno: int
+    string: bool
```

Comment:
I'm not in love with this name, as if we add `try/except` support then "string" isn't correct.

Reply:
I like `weak` a lot - there's prior art with languages like Rust using `WeakReference`.

It'd be great to add docstring explaining what is meant by "weak".

--

Alternatively, what about using an enum? Do you think we would want to distinguish between unconditional import vs string import vs try/except vs something else? That might be premature, I'm curious what you think.

Reply:
I'm not against an Enum, but likely premature. Also an enum means I need to come up with _more_ names.

Reply:
I'm not against an Enum, but likely premature. Also an enum means I need to come up with _more_ names. So I'll probably punt for now :sweat_smile:

Reply:
Actually, maybe an Enum would help being explicit in the tests :thinking:  I'll try it

Reply:
And now I'm floundering on naming again :sweat: "weak" vs. "strong" it is!

Reply:
Yeah I wouldn't distinguish between "try/except" yet though, as that would be a lie. Reminder that we can rename this Enum anytime we want - this code isn't really part of the Plugin API imo.

Some other names to consider:
* conventional / string
* normal / string

Reply:
Yeah I'm not mentioning try/except _yet_. Will update ASAP I think using `ParsedPythonImportInfo` in the test works out well.